testhelper/tests.go

```diff
@@ -231,121 +267,189 @@ func (options *TestOptions) RunTestUpgrade() (*terraform.PlanStruct, error) {
+		logger.Log(options.Testing, "Init / Apply on Base branch dir:", options.TerraformOptions.TerraformDir)
 
 		_, resultErr = terraform.InitAndApplyE(options.Testing, options.TerraformOptions)
-		assert.Nilf(options.Testing, resultErr, "Terraform Apply on MASTER branch has failed")
-
-		// Get the path to the state file
-		statePath := path.Join(options.TerraformDir, "terraform.tfstate")
-
-		// Copy the state file to the temporary directory
-		errCopyState := common.CopyFile(statePath, path.Join(tempDir, "terraform.tfstate"))
-
-		// Only proceed to upgrade Test of master branch apply passed
-		if resultErr == nil && assert.Nil(options.Testing, errCopyState, fmt.Sprintf("Error copying state: %s", errCopyState)) {
-
-			logger.Log(options.Testing, "Attempting Git Checkout PR Branch:", prRef.Name(), "-", prRef.Hash())
-			// checkout the HASH of original (PR) branch.
-			// NOTE: in automation the original checkout branch is detached and points to the pseudo merge of the PR.
-			//       These detached merge branches report their name as "HEAD" which is not a suitable checkout value.
-			//       The solution here is to do this final checkout on the HASH of the original branch which will work
-			//       with both detached and normal branches.
-			resultErr = w.Checkout(&git.CheckoutOptions{
-				Hash:  prRef.Hash(),
-				Force: true})
-			assert.Nilf(options.Testing, resultErr, "Could Not Checkout PR Branch")
-
-			// Copy the state file back to the original location
-			errCopyState := common.CopyFile(path.Join(tempDir, "terraform.tfstate"), statePath)
-
-			if resultErr == nil && assert.Nil(options.Testing, errCopyState, fmt.Sprintf("State file not found cannot compare plan for test\nError copying state: %s", errCopyState)) {
-				cur, _ = gitRepo.Head()
-				logger.Log(options.Testing, "Current Branch (PR):", cur.Name(), "-", cur.Hash())
-
-				result, resultErr = options.runTestPlan()
-				assert.Nilf(options.Testing, resultErr, "Terraform Plan on PR branch has failed")
-				if result != nil && resultErr == nil {
-					logger.Log(options.Testing, "Parsing plan output to determine if any resources identified for destroy (PR branch)..")
-					options.checkConsistency(result)
-					// Adding optional upgrade support on PR Branch
-					if options.CheckApplyResultForUpgrade && !options.Testing.Failed() {
-						logger.Log(options.Testing, "Validating Optional upgrade on Current Branch (PR):", cur.Name())
-						_, resultErr = terraform.ApplyE(options.Testing, options.TerraformOptions)
-						assert.Nilf(options.Testing, resultErr, "Terraform Apply on PR branch has failed")
-					}
-				} else {
-					// if there were issues running InitAndPlan, an Init needs to take place after branch change in order for downstream
-					// terraform to work (like the destroy)
-					terraform.Init(options.Testing, options.TerraformOptions)
-				}
+		assert.Nilf(options.Testing, resultErr, "Terraform Apply on Base branch has failed")
```

Comment:
If base apply has failed, should we continue? assert() is only going to mark the test as failed but processing will continue, which might not be necessary if base fails.